app/uk/gov/hmrc/gform/lookup/ChoiceRuntimeIndexService.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.lookup
+
+import org.apache.lucene.analysis.standard.StandardAnalyzer
+import org.apache.lucene.document.{ Document, Field, StringField, TextField }
+import org.apache.lucene.index.{ DirectoryReader, IndexWriter, IndexWriterConfig, Term }
+import org.apache.lucene.search.{ BooleanQuery, IndexSearcher, PrefixQuery }
+import org.apache.lucene.search.BooleanClause.Occur
+import org.apache.lucene.store.ByteBuffersDirectory
+import play.api.libs.json.{ Json, OFormat }
+
+import scala.util.{ Failure, Success, Try }
+
+case class ChoiceOption(value: String, label: String, keyWord: Option[String] = None)
+case class ChoiceSearchResult(value: String, label: String)
+
+object ChoiceSearchResult {
+  implicit val format: OFormat[ChoiceSearchResult] = Json.format[ChoiceSearchResult]
+}
+
+class ChoiceRuntimeIndexService {
+
+  private val analyzer = new StandardAnalyzer()
```

Comment:
Let's not use `StandardAnalyzer`. When label contains for example `$` you can search by it.

We should use [WhiteSpaceOnlyAnalyzer](https://github.com/hmrc/gform-frontend/blob/aea11804f02b4eb29a1086eb2f6e56b9f7f9fe57/project/LuceneIndexCreator.scala#L96-L102) like  other lookups.